dpgen2/op/run_dp_train.py

```diff
@@ -190,12 +190,14 @@ def write_data_to_input_script(
         if major_version == "1":
             # v1 behavior
             odict['training']['systems'] = data_list
-            odict['training']['batch_size'] = "auto"
+            if odict['training'].get('batch_size') is None:
+                odict['training']['batch_size'] = "auto"
             odict['training']['auto_prob_style'] = auto_prob_str
         elif major_version == "2":
             # v2 behavior
             odict['training']['training_data']['systems'] = data_list
-            odict['training']['training_data']['batch_size'] = "auto"
+            if odict['training']['training_data'].get('batch_size') is None:
+                odict['training']['training_data']['batch_size'] = "auto"
```

Comment:
Use [`dict.setdefault`](https://docs.python.org/3/library/stdtypes.html#dict.setdefault).
```suggestion
            odict['training']['training_data'].setdefault('batch_size', "auto")
```